docs/virtual_machines/accessing_virtual_machines.md

```diff
@@ -290,21 +218,168 @@ spec:
+          image: quay.io/containerdisks/fedora:latest
+        name: containerdisk
       - cloudInitConfigDrive:
-          userData: |
-            #!/bin/bash
+          userData: |-
+            #cloud-config
+            password: fedora
+            chpasswd: { expire: False }
+            # Disable SELinux for now, so qemu-guest-agent can write the authorized_keys file
+            # The selinux-policy is too restrictive currently, see open bugs:
+            #   - https://bugzilla.redhat.com/show_bug.cgi?id=1917024
+            #   - https://bugzilla.redhat.com/show_bug.cgi?id=2028762
+            #   - https://bugzilla.redhat.com/show_bug.cgi?id=2057310
+            bootcmd:
+              - setenforce 0
+        name: cloudinitdisk
+EOF
+```
+
+### Accessing the VMI using virtctl
+
+The user can create a websocket backed network tunnel to a port inside the
+instance by using the `virtualmachineinstances/portforward` subresource of the
+`VirtualMachineInstance`.
+
+One use-case for this subresource is to forward SSH traffic into
+the `VirtualMachineInstance` either from the CLI or a web-UI.
 
-            sudo setenforce Permissive
-            sudo yum install -y qemu-guest-agent
-            sudo systemctl start qemu-guest-agent
-        name: disk1
-END
+To connect to a `VirtualMachineInstance` from your local machine, `virtctl`
+provides a lightweight SSH client with the `ssh` command, that uses port
+forwarding. Refer to the command's help for more details.
 
-kubectl create -f my-vm.yaml
+```shell
+virtctl ssh
+```
+
+To transfer files from or to a `VirtualMachineInstance` `virtctl` also
+provides a lightweight SCP client with the `scp` command. Its usage is
+similar to the `ssh` command. Refer to the command's help for more details.
+
+```shell
+virtctl scp
+```
+
+#### Using virtctl as proxy
+
+If you prefer to use your local OpenSSH client, there are two ways of doing
+that in combination with virtctl.
+
+> Note: Most of this applies to the `virtctl scp` command too.
+
+1. The `virtctl ssh` command has a `--local-ssh` option. With this option
+   `virtctl` wraps the local OpenSSH client transparently to the user. The
+   executed SSH command can be viewed by increasing the verbosity (`-v 3`).
+
+```shell
+virtctl ssh --local-ssh -v 3 testvm
+```
+
+2. The `virtctl port-forward` command provides an option to tunnel a single
+   port to your local stdout/stdin. This allows the command to be used in
+   combination with the OpenSSH client's `ProxyCommand` option.
+
+```shell
+ssh -o 'ProxyCommand=virtctl port-forward --stdio=true vmi/testvm.mynamespace 22' fedora@testvm.mynamespace
+```
+
+To provide easier access to arbitrary virtual machines you can add the following
+lines to your SSH `config`:
+
+```
+Host vmi/*
+   ProxyCommand virtctl port-forward --stdio=true %h %p
+Host vm/*
+   ProxyCommand virtctl port-forward --stdio=true %h %p
+```
+
+This allows you to simply call `ssh user@vmi/testvmi.mynamespace` and your
```

Comment:
Maybe in a follow-up we can highlight how to use a ssh config file in combination with ansible.

Something like:

```
[ssh_connection]
ssh_args = -F ~/.ssh/myconfig
```

Then people can write their inventory files like always and provision VMs over the mgmt-network :)

Reply:
Good call, I added a small paragraph describing this.